### ISSUE TITLE
Unstable result on training set with same code

### ISSUE THREAD

**Issue by JonathanChen1002** [microsoft/SynapseML#1179]
**Problem**

I'm doing a binary classification task with LightGBMClassifier.  In most cases(90% ~ 95% of experimental run), the precision in top 20% of training set could achieve 0.5. But in few cases, the precision drop to 0.3(which is close to original distribution - 391021 negative and 117095 positive)

**Version**

- mmlspark: 1.0.0-rc3-49-659b7743-SNAPSHOT
- Spark: 3.1.1 (on Databricks 8.3)

**Model**

`LightGBMClassifier(labelCol="label", featuresCol="features", categoricalSlotNames = cat_cols, useBarrierExecutionMode=True, numIterations=300)`

I'm not sure the unstable result is due to randomness of model(in this case, is there a way to ensure same result for every run?) or just a bug. Thanks.

**Comment by imatiach-msft** [comment]
closing as lightgbm is now deterministic with merged PR https://github.com/microsoft/SynapseML/pull/1387 as long as seed and deterministic=True parameters are set
